### ISSUE TITLE
`import` from `process` left behind in browser-targeted output

### ISSUE THREAD

**Issue by mindplay-dk** [parcel-bundler/parcel#7904]
# 🐛 bug report

I'm trying to build a "universal" ES6 library - meaning, it should be able to load in modern Node, modern browsers, and probably Deno too.

The addition of `browserslist` appears to resolve all `process.env` references statically - however, one reference looks like it was left behind, and the resulting module leads with:

    import {platform as $gh3jN$platform, env as $gh3jN$env} from "process";

Shouldn't there be an error or warning for this, when the bundler knows it's building for a browser context, and the resulting build still has a Node specific `import` left in it?

It appears the lingering `process` import comes from the `ansi-colors`, which [contains this](https://github.com/doowb/ansi-colors/blob/a4794363369d7b4d1872d248fc43a12761640d8e/index.js#L13-L15):

```js
  if ('FORCE_COLOR' in process.env) {
    colors.enabled = process.env.FORCE_COLOR !== '0';
  }
```

Perhaps the `in` operator is not supported by static analysis?

Or maybe the static analysis is getting confused by the presence of two nested tests?

## 🎛 Configuration (.babelrc, package.json, cli command)

My attempted `package.json` (and the project) is located [here](https://github.com/mindplay-dk/inertion/blob/web-compat/package.json).

## 🤔 Expected Behavior

Probably at least a warning or error, for either one or both of these:

1. Unable to remove all `process` references from a browser target.
2. Unable to handle a `process.env` reference via static analysis for any reason.

Ideally, static analysis should probably be enhanced to support this though? It looks like both `rollup` and `webpack` are capable of handling this case, so it would probably be best to have parity with that, as it's rather influential on the ecosystem already.

## 😯 Current Behavior

(see above)

## 💁 Possible ~~Solution~~ Workaround

I could submit a PR to `ansi-colors` and try to make it more compatible with Parcel's static analysis. (assuming it's possible to somehow rewrite that `process.env` check in a way that works the same?)

## 🔦 Context

(see project referenced above.)

## 💻 Code Sample

(see references above.)

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | `parcel@2.4.1`, `@parcel/packager-ts@2.4.1`, `@parcel/transformer-typescript-types@2.4.1`
| Node             | `v16.14.2`
| npm/Yarn         | n/a
| Operating System | Ubuntu 16.04.7 LTS, (Release:        16.04, Codename:       xenial)

**Comment by mischnic** [comment]
>   if ('FORCE_COLOR' in process.env)

We could definitely support that: https://github.com/parcel-bundler/parcel/blob/v2/packages/transformers/js/core/src/env_replacer.rs

**Comment by mindplay-dk** [comment]
Unfortunately, I don't know Rust...

**Comment by mindplay-dk** [comment]
I've tested the `2.5.0` release, and it still leaves behind the `platform` import.

    import{platform as e}from"process";

I've pushed it to a branch [here](https://github.com/mindplay-dk/inertion/tree/parcel-ansi-colors).

Apparently it's due to these references in `ansi-colors` [here](https://github.com/doowb/ansi-colors/blob/a4794363369d7b4d1872d248fc43a12761640d8e/symbols.js#L4-L5):

```
const isWindows = process.platform === 'win32';
const isLinux = process.platform === 'linux';
```

Is it supposed to handle `platform` as well as `env`?

(Should I file a separate issue for this?)
